### PR TITLE
Feat/net/controlplane Manage FailureDomain on multipld az

### LINKED ISSUE #287: [Feature]:  Manage ControlPlane déployment on Multi AZ
### Explain problem to solve

We need to deploy ControlPlane replicas on Multi Availability Zone (Sub region) to prevent SubRegion Failure

### Describe the solution you would like

We need to use the FailureDomains capability as already implemented by [AWS](https://cluster-api-aws.sigs.k8s.io/topics/failure-domains/index.html) and [Azure](https://capz.sigs.k8s.io/topics/failure-domains#:~:text=The%20Cluster%20API%20controller%20will,across%20all%20the%20failure%20domains.) providers

### Additional context

this feature can be extended to deployments

### Environment

```markdown
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):
- Kernel (e.g. `uname -a`):
- cluster-api-provider-outscale version:
- cluster-api version:
- Install tools:
- Kubernetes Distribution:
- Kubernetes Diestribution version:
```

### PR DESCRIPTION
**Pull request by sebglon**
**What type of PR is this?**

/kind feature           New functionality with test.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #287,  #284 

**Special notes for your reviewer**:
related documentation: https://github.com/kubernetes-sigs/cluster-api/issues/2131
DEPENDENCIES:
https://github.com/outscale/cluster-api-provider-outscale/pull/276

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

**Comment by gvdhart** [comment]
DEPENDENCIES:
https://github.com/outscale/cluster-api-provider-outscale/pull/276

**Comment by outscale-vbr** [comment]
@sebglon @gvdhart 
Is that PR ready ?
Please rebase now from main ?

**Comment by sebglon** [comment]
@outscale-vbr PR is ready

File: controllers/oscmachine_vm_controller_unit_test.go [hunk 1]
```diff
@@ -3653,9 +3817,11 @@ func TestReconcileDeleteVm(t *testing.T) {
 				EXPECT().
 				DeleteVm(gomock.Eq(vmId)).
 				Return(vtc.expDeleteVmErr)
+
 			reconcileDeleteVm, err := reconcileDeleteVm(ctx, clusterScope, machineScope, mockOscVmInterface, mockOscPublicIpInterface, mockOscLoadBalancerInterface, mockOscSecurityGroupInterface)
 			if err != nil {
-				assert.Equal(t, vtc.expReconcileDeleteVmErr.Error(), err.Error(), "reconccileDeleteVm() hould return the same error")
+				t.Logf(err.Error())
+				assert.Equal(t, vtc.expReconcileDeleteVmErr.Error(), err.Error(), "reconccileDeleteVm() should return the same error")
```
[the remark this answers is not in the document]
Comment (sebglon) [reply controllers/oscmachine_vm_controller_unit_test.go]: done

File: controllers/oscmachine_vm_controller_unit_test.go [hunk 2]
```diff
@@ -3525,6 +3687,7 @@ func TestReconcileDeleteVm(t *testing.T) {
 			name:                                    "failed to delete vm",
 			clusterSpec:                             defaultVmClusterReconcile,
 			machineSpec:                             defaultVmReconcile,
+			expListMachine:                          false,
```
Comment (outscale-vbr) [inline controllers/oscmachine_vm_controller_unit_test.go:3690]: Duplicate

**Comment by outscale-vbr** [comment]
Ok

**Review by outscale-vbr — APPROVED** [review]

**Pull request merged**